### PR TITLE
[Feature] [SR SQL Planner Summer Camp] add createRepository analyzer

### PR DESCRIPTION
**Pull request by melt-code**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7951

## Problem Summary(Required) ：

add createRepository analyzer

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 45 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[418]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|1|2|50.00%|[113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|20|27|74.07%|[16, 32, 38, 44, 49, 50, 55]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|13|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 38 / 48 (79.17%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[443]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|1|2|50.00%|[113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[943]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|20|27|74.07%|[15, 31, 37, 43, 48, 49, 54]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|13|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 39 / 48 (81.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[443]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[943]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|20|27|74.07%|[15, 31, 37, 43, 48, 49, 54]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -836,6 +840,14 @@ showProcStatement
     : SHOW PROC path=string
     ;
 
+// ---------------------------------------- Backup Store Statement -----------------------------------------------------
+createRepositoryStatement
+    : CREATE (READ ONLY)? REPOSITORY identifier
+    WITH BROKER identifier
+    ON 'LOCATION' string
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:847]: Move 'LOCATION' to StarRocksLex.g4, and add it in nonReserved section

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -136,6 +136,10 @@ statement
 
     // proc
     | showProcStatement                                                                      #showProc
+
+    //  Backup Store Satement
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:140]: error comment

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 45 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[456]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[953]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|20|27|74.07%|[15, 31, 37, 43, 48, 49, 54]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|2|2|100.00%|[]|

**Comment by melt-code** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 45 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[422]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[953]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|20|27|74.07%|[15, 31, 37, 43, 48, 49, 54]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 1 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4]: It has been amended

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 2 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4]: It has been amended

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 45 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[422]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[953]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|20|27|74.07%|[15, 30, 36, 42, 47, 48, 53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 45 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[468]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[993]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|20|27|74.07%|[15, 30, 36, 42, 47, 48, 53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|2|2|100.00%|[]|

**Comment by xiaoyong-z** [comment]
@melt-code now we support backup and restore supports without broker: https://github.com/StarRocks/starrocks/pull/9319. 
New createRepository analyzer here should also support this.

**Comment by xiaoyong-z** [comment]
Grammar for create repository without broker 

image

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 48 / 55 (87.27%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[480]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[1015]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|23|28|82.14%|[15, 30, 47, 48, 53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|2|2|100.00%|[]|

**Comment by melt-code** [comment]
> Grammar for create repository without broker
> 
> image

Syntax compatibility for this grammar has been added

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 41 / 48 (85.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[453]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[1269]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateRepositoryAnalyzer.java|23|28|82.14%|[15, 30, 47, 48, 53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|2|2|100.00%|[]|

**Comment by melt-code** [comment]
run starrocks_admit_test

**Review by HangyuanLiu — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**